### PR TITLE
test driver config

### PR DESCRIPTION
**Pull request by pohly**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:

We want to get rid of hostpath driver specific code in csi-release-tools, both to simplify it and to make it applicable to other CSI drivers.

**Which issue(s) this PR fixes**:
Related-to https://github.com/kubernetes-csi/csi-release-tools/issues/32

**Special notes for your reviewer**:

This needs to be rolled out like this:
- merge this PR into csi-driver-host-path
- do a 1.3.0-rc release
- use that release as `CSI_PROW_DRIVER_REVISION` in prow.sh and update csi-release-tools
- merge that change into all repos

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

### CODE REVIEW

**Comment by pohly** [comment]
Testing with the modified prow.sh worked:
- https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/kubernetes-csi_csi-driver-host-path/143/pull-kubernetes-csi-csi-driver-host-path-1-15-on-kubernetes-1-15/1226842527286431744
- https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/kubernetes-csi_csi-driver-host-path/143/pull-kubernetes-csi-csi-driver-host-path-1-16-on-kubernetes-1-16/1226842527290626048
- https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/kubernetes-csi_csi-driver-host-path/143/pull-kubernetes-csi-csi-driver-host-path-1-17-on-kubernetes-1-17/1226842527290626049

It also worked for the external-provisioner, see https://github.com/kubernetes-csi/external-provisioner/pull/403.

I've moved the csi-release-tools changes to https://github.com/kubernetes-csi/csi-release-tools/pull/69.

File: deploy/kubernetes-1.17/test-driver.yaml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+StorageClass:
+  FromName: true
+SnapshotClass:
+  FromName: true
+DriverInfo:
+  Name: hostpath.csi.k8s.io
+  Capabilities:
+    block: true
+    controllerExpansion: true
```
Comment (msau42) [inline deploy/kubernetes-1.17/test-driver.yaml]: @gnufied were resizing tests added in 1.16 or 1.17?

Comment (pohly) [reply deploy/kubernetes-1.17/test-driver.yaml]: According to `git diff v1.15.0..v1.16.0 test/e2e/storage/testsuites/testdriver.go`, `controllerExpansion` was added in 1.16:
```
@@ -126,7 +152,10 @@ const (
        // - NodeStageVolume in the spec
        CapMultiPODs Capability = "multipods"
 
-       CapRWX Capability = "RWX" // support ReadWriteMany access modes
+       CapRWX                 Capability = "RWX"                 // support ReadWriteMany access modes
+       CapControllerExpansion Capability = "controllerExpansion" // support volume expansion for controller
+       CapNodeExpansion       Capability = "nodeExpansion"       // support volume expansion for node
+       CapVolumeLimits                   = "volumeLimits"        // support volume limits (can be *very* slow)
 )
```

Comment (msau42) [reply deploy/kubernetes-1.17/test-driver.yaml]: Shall we enable the tests in 1.16 too then?

Comment (pohly) [reply deploy/kubernetes-1.17/test-driver.yaml]: What's missing for them in 1.16?

Comment (msau42) [reply deploy/kubernetes-1.17/test-driver.yaml]: I don't see the capabilities enabled in the 1.16 config

Comment (pohly) [reply deploy/kubernetes-1.17/test-driver.yaml]: Oh, right. I that the change above was for 1.16, but it's 1.17. Let me add it for 1.16 then, too.

Comment (pohly) [reply deploy/kubernetes-1.17/test-driver.yaml]: Revised test-driver.yaml files:
```
$ diff deploy/kubernetes-1.15/test-driver.yaml deploy/kubernetes-1.16/test-driver.yaml
2,3c2,3
< # using the Kubernetes 1.15 E2E test suite. For details see:
< # https://github.com/kubernetes/kubernetes/tree/v1.15.0/test/e2e/storage/external
---
> # using the Kubernetes 1.16 E2E test suite. For details see:
> # https://github.com/kubernetes/kubernetes/tree/v1.16.0/test/e2e/storage/external
12a13
>     controllerExpansion: true
14d14
<     dataSource: true
15a16
>     nodeExpansion: true
16a18
>     snapshotDataSource: true

$ diff deploy/kubernetes-1.16/test-driver.yaml deploy/kubernetes-1.17/test-driver.yaml
2,3c2,3
< # using the Kubernetes 1.16 E2E test suite. For details see:
< # https://github.com/kubernetes/kubernetes/tree/v1.16.0/test/e2e/storage/external
---
> # using the Kubernetes 1.17 E2E test suite. For details see:
> # https://github.com/kubernetes/kubernetes/tree/v1.17.0/test/e2e/storage/external
17a18
>     singleNodeVolume: true
18a20
>     topology: true
```

**Comment by pohly** [comment]
@msau42: I also pulled the corresponding release-tools update into this PR, so the test runs really were done with the new `test-driver.yaml` files.

**Pull request merged**